.github/workflows/ssl_certificate_renewal.yaml

```
@@ -0,0 +1,236 @@
+
+      - name: Fetch and check certificate
+        id: fetch_and_check
+        if: inputs.force_renew_cert == false
+        env:
+          string_manipulation_cert_name: ${{ inputs.csr_common_name }}
+          key_vault_name: ${{ inputs.key_vault_name }}
+        run: |
+            key_vault_cert_name=$(echo "${string_manipulation_cert_name//./-}")
+
+            if ! az keyvault certificate download --vault-name "$key_vault_name" -n "$key_vault_cert_name" -f certificate.pem >/dev/null
+              then
+                echo "❌ The certificate was not found in the selected Key Vault, proceeding to request it from LE"
+                echo "require_new_certificate=true" >> $GITHUB_OUTPUT
+            else
+              if ! openssl x509 -inform pem -checkend "$certificate_expiration_time" -noout -in certificate.pem
+                then
+                  # certificate is expiring or is already expired
+                  echo "require_new_certificate=true" >> $GITHUB_OUTPUT
+                else
+                  echo "✅ The certificate is not expiring, cleaning up and exiting"
+              fi
+              rm -f certificate.pem
+            fi
+
+      - name: Set renew_certificate
+        id: should_renew
+        run: |
+          if [[ "$FORCE_RENEW_CERT" == "true" ]]; then
+            echo "renew_certificate=true" >> $GITHUB_OUTPUT
+            echo "## ⚠️ Renewal Status: FORCED" >> $GITHUB_STEP_SUMMARY
+            echo "**Renewal explicitly requested.** The renewal action will proceed. 🛠️" >> $GITHUB_STEP_SUMMARY
+          elif [[ "$REQUIRE_NEW_CERT" == "true" ]]; then
+            echo "renew_certificate=true" >> $GITHUB_OUTPUT
+            echo "## 🔑 Renewal Status: REQUIRED" >> $GITHUB_STEP_SUMMARY
+            echo "**Verification successful.** The certificate will be renewed in the next step. ✅" >> $GITHUB_STEP_SUMMARY
+          else
+            echo "renew_certificate=false" >> $GITHUB_OUTPUT
+            echo "## ✅ Renewal Status: NOT NEEDED" >> $GITHUB_STEP_SUMMARY
+            echo "No certificate renewal is required at this time. 🟢" >> $GITHUB_STEP_SUMMARY
+          fi
+        env:
+          FORCE_RENEW_CERT: ${{ inputs.force_renew_cert }}
+          REQUIRE_NEW_CERT: ${{ steps.fetch_and_check.outputs.require_new_certificate }}
+
+      - name: Fetch LE authentication keys from Key Vault
+        if: steps.should_renew.outputs.renew_certificate == 'true'
+        env:
+          key_vault_name: ${{ inputs.key_vault_name }}
+        run: |
+            az keyvault secret download --vault-name "$key_vault_name" --name le-private-key-json --file private_key.json
+            az keyvault secret download --vault-name "$key_vault_name" --name le-regr-json --file regr.json
+
+      - name: Run LE renew SSL Certificate python action
+        if: steps.should_renew.outputs.renew_certificate == 'true'
+        uses: pagopa/io-infra/.github/actions/le-renew-ssl-certificate-python@IOPLT-1309-ssl-cert-renewal-template
```

```suggestion
        uses: pagopa/io-infra/.github/actions/le-renew-ssl-certificate-python@main
```

**reminder**: Change to **main** before merge